### PR TITLE
Add new encoding "UTF-8 without BOM"

### PR DESCRIPTION
**Pull request by niksedk**
Notepad and Notepad++ (now) have two different encodings for UTF-8 - with and without BOM, and there seems to be more and more requests about this - like #3972.

This PR will rename the old "UTF-8" to "UTF-8 with BOM" and add a new encoding to the encodings combo-box in the UI:
image
Also, the "secret" setting `WriteUtf8Bom` has been removed.

ATM, BOM is not really part of the `Encoding` object in C#, so it requires a bit more care than just an normal `Encoding`.

### CODE REVIEW

File: src/Logic/CommandLineConvert/CommandLineConverter.cs [hunk 1]
```diff
@@ -219,13 +219,31 @@ private static int Help(string[] arguments)
...
+                    {
+                        targetEncoding = new TextEncoding(Encoding.UTF8, TextEncoding.Utf8WithBom);
+                    }
+                    else if (encodingName.Equals("utf-8-no-bom", StringComparison.OrdinalIgnoreCase) ||
+                             encodingName.Equals(TextEncoding.Utf8WithoutBom.Replace(" ", string.Empty), StringComparison.OrdinalIgnoreCase) ||
+                             encodingName.Equals(TextEncoding.Utf8WithoutBom.Replace(" ", "-"), StringComparison.OrdinalIgnoreCase))
+                    {
+                        targetEncoding = new TextEncoding(Encoding.UTF8, TextEncoding.Utf8WithoutBom);
+                    }
                     if (encodingName.Length > 0)
```
Comment (xylographe) [inline src/Logic/CommandLineConvert/CommandLineConverter.cs]:
```
else if (encodingName.Length > 0)
```

**Comment by OmrSi** [comment]
@niksedk Is the default without BOM?

**Comment by niksedk** [comment]
Yes, the default is with BOM as it has always been (you can change the default in Option - Settings though)

**Comment by OmrSi** [comment]
Cool. It must have been a problem with my settings, it defaulted to "without BOM" for some reason.
Thank you.

**Comment by niksedk** [comment]
Oh, if you open a file already in "UTF-8 wihtout BOM" that encoding will be kept.

**Pull request merged**